internal/integrations/k8s/k8s.go

```diff
@@ -0,0 +1,43 @@
+package k8s
+
+import (
+	"ship-it/internal/models"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	kv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+type K8sInterface interface {
+	ConfigMaps(namespace string) kv1.ConfigMapInterface
+	ListAll() (*v1.ConfigMapList, error)
+}
+
+func (k *K8sClient) ListAll() (*v1.ConfigMapList, error) {
+	return k.ConfigMaps(v1.NamespaceAll).List(metav1.ListOptions{})
+}
+
+type K8sClient struct {
+	kv1.CoreV1Interface
+}
+
+func NewK8sClient() (K8sInterface, error) {
```

Comment:
We should avoid stuttering (`k8s.NewK8sClient`) and name the constructor `New` instead.